### DISCUSSION TITLE
Brreeze theme compatibiltiy with price calculator module

### DISCUSSION THREAD

**Discussion by bhawna117** [breezefront/community#211] [3. Integrations]
Hi , there 

I am using the Magecomp Price calculator with the Breeze theme but this is not working. i have this code in requirejs-config.js

```js
var config = {
    deps: ['Magento_Catalog/js/price-utils','Magecomp_Pricecalculator/js/flooring-product-view'],
    map: {
        '*': {
            FlooringProductInit: 'Magecomp_Pricecalculator/js/flooring-product-view',
            priceUtils: 'Magento_Catalog/js/price-utils',
        }
    },
    config: {
        mixins: {
            'Magento_ConfigurableProduct/js/configurable': {
                'Magecomp_Pricecalculator/js/model/skuswitch': true
            },
            'Magento_Swatches/js/swatch-renderer': {
                'Magecomp_Pricecalculator/js/model/swatch-skuswitch': true
            }
        }
    }
}
```

 and this is not working when I enable the breeze theme. Please guide how can I make it compatible with breeze.

**Comment by vovayatsyuk** [comment]
You can try to integrate it if you have JS experience. 

It seems that `deps` section is useless in your case. You can skip it. `priceUtils` in `map` section is useless too. You can skip it too.

So, you need to integrate two mixins and one widget. Here are some guides we have:

- https://breezefront.com/docs/integration-tips
- https://breezefront.com/docs/integration-example

**Comment by bhawna117** [comment]
I have integrated the mixin, for widget which section do you mean? map for ?

**Reply by vovayatsyuk** [reply]
This one: Magecomp_Pricecalculator/js/flooring-product-view

**Comment by bhawna117** [comment]
BUt there is confusion this one is used as mixin too that is why I am
confused so could you let me know how to achieve this.

On Thu, Nov 23, 2023 at 3:17 PM Vova Yatsyuk ***@***.***>
wrote:

> This one: Magecomp_Pricecalculator/js/flooring-product-view
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/breezefront/discussions/211#discussioncomment-7650179>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AN2PQPF7CU5MWFSWJVN32U3YF4LSLAVCNFSM6AAAAAA7WA3JYGVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TMNJQGE3TS>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Thanks & Regards
Bhawna

**Reply by vovayatsyuk** [reply]
Sorry, I don't understand. You need to create two mixins (Magecomp_Pricecalculator/js/model/skuswitch, Magecomp_Pricecalculator/js/model/swatch-skuswitch) and one component (Magecomp_Pricecalculator/js/flooring-product-view).

**Comment by bhawna117** [comment]
and to do this I need to add these files from breeze_default.xml like this
, ?
<?xml version="1.0"?>
<page xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"

xsi:noNamespaceSchemaLocation="urn:magento:framework:View/Layout/etc/page_configuration.xsd">
    <body>
       <referenceBlock name="breeze.js">
        <arguments>
            <argument name="bundles" xsi:type="array">
                <item name="default" xsi:type="array">
                    <item name="items" xsi:type="array">
                        
                        <item name="module-flooring-product-view"
xsi:type="string">Magecomp_Pricecalculator/js/flooring-product-view</item>
                         <item name="skuswitch"
xsi:type="string">Magecomp_Pricecalculator/js/skuswitch</item>
                         <item name="swatch-skuswitch"
xsi:type="string">Magecomp_Pricecalculator/js/swatch-skuswitch</item>
                    </item>
                </item>

            </argument>
        </arguments>
        </referenceBlock>
    </body>
</page>

and then added mixin for example for this file skuswitch.js
define([
    'jquery',
    'mage/utils/wrapper'
], function ($, wrapper) {
    'use strict';
    $.mixin('Magento_Swatches/js/swatch-renderer', {
        create: function(targetModule){
                var reloadPrice = targetModule.prototype._reloadPrice;
                var reloadPriceWrapper = wrapper.wrap(reloadPrice,
function(original){
                    var result = original();
                    var simpleSku =
this.options.spConfig.skus[this.simpleProduct];
                    if(this.simpleProduct != null){
                        localStorage.setItem("currentproduct",
this.simpleProduct);
                    }
                    if(simpleSku != '') {
                        $('div.product-info-main .sku
.value').html(simpleSku);
                    }
                    return result;
                });
                targetModule.prototype._reloadPrice = reloadPriceWrapper;
                //return targetModule;
                targetModule();
            }
    });
});

Let me know if I am doing wrong ? Or do I do not need to add them from
breeze_default.xml ?

On Thu, Nov 23, 2023 at 3:25 PM Vova Yatsyuk ***@***.***>
wrote:

> Sorry, I don't understand. You need to create two mixins
> (Magecomp_Pricecalculator/js/model/skuswitch,
> Magecomp_Pricecalculator/js/model/swatch-skuswitch) and one component
> (Magecomp_Pricecalculator/js/flooring-product-view).
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/breezefront/discussions/211#discussioncomment-7650256>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AN2PQPF5KSRXCSFCNQT722DYF4MSFAVCNFSM6AAAAAA7WA3JYGVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TMNJQGI2TM>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Thanks & Regards
Bhawna

**Reply by vovayatsyuk** [reply]
Yes. 

But I have a few questions about your code:

 - In the mixin declaration you've used component path instead of name: `$.mixin('Magento_Swatches/js/swatch-renderer', {`. The correct declaration is: `$.mixin('SwatchRenderer', {`
- I don't understand why you override `create` method to override `_reloadPrice`
- The is no `_reloadPrice` method in `Magento_Swatches/js/swatch-renderer`. Did you mean [_UpdatePrice](https://github.com/breezefront/module-breeze/blob/master/view/frontend/web/js/components/swatch-renderer.js#L886)?

If I understand your code correctly you wanted to do the following:

```js
$.mixin('SwatchRenderer', {
    _UpdatePrice: function (original) {
        // overridden logic
    }
});
```

**Comment by bhawna117** [comment]
okay thank you for the clarification I have corrected that and could you
help me with component too Here is hte file code for that

```js
define([
    "jquery",
    'priceUtils'
], function ($, priceUtils) {
    'use strict';

    var priceFormat = {
        decimalSymbol: '.'
    };
    var FlooringProductInit = {
        FlooringProductInit: function (config) {
            this.config = config;
            ...
        },
        ...
    };

    return FlooringProductInit;
});
```

**Comment by bhawna117** [comment]
can youplease let me know I have updated the above file with this

```js
$.view('FlooringProductInit', {
    component: 'FlooringProductInit',
    defaults: {
        template: 'Magecomp_Pricecalculator/flooring/init'
    },
    create: function () { alert('dfd');
        var priceFormat = {
            decimalSymbol: '.'
        };
        var FlooringProductInit = {
            FlooringProductInit: function (config) {
                this.config = config;
                ...
            },
            ...
        };

        return FlooringProductInit;
    }
});
```

and in phtml file calling this way

```html
  <script type="text/x-magento-init">
    {
        "*": {
            "FlooringProductInit": <?= $config ?>
        }
    }
    </script>
```

but this is not working

**Comment by bhawna117** [comment]
this is just the file code, I just want to know if my initial syntax is
fine example
$.view('FlooringProductInit', {
    component: 'FlooringProductInit',
    defaults: {
        template: 'Magecomp_Pricecalculator/flooring/init'
    },
    create: function () { alert('dfd3');

}
});

On Thu, Nov 23, 2023 at 6:17 PM Vova Yatsyuk ***@***.***>
wrote:

> Hey man, that's way too much code for me)) Please provide the minimum code
> for your issue.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/breezefront/discussions/211#discussioncomment-7651616>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AN2PQPEZH7NH6CWKLLCWTUTYF5AV5AVCNFSM6AAAAAA7WA3JYGVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TMNJRGYYTM>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Thanks & Regards
Bhawna

**Reply by vovayatsyuk** [reply]
Why do you use $.view? Where did you get the template?

$.view - is a uiComponent replacement, but original code using plain object. There is no template in original code also..

**Comment by bhawna117** [comment]
so how should Inise it?

**Reply by vovayatsyuk** [reply]
Just add the `component: 'FlooringProductInit'` line.

```
var FlooringProductInit = {
    component: 'FlooringProductInit',
    FlooringProductInit: function (config) {
```

Source: https://breezefront.com/docs/integration-tips#object
